apps/nrm/behaviours/are-you-sure.js

```diff
@@ -20,18 +21,19 @@ module.exports = superclass => class extends superclass {
     });
   }
 
-  saveValues(req, res, next) {
+  async saveValues(req, res, next) {
     const id = req.body.confirm;
-
-    if (id) {
-      const email = encodeEmail(req.sessionModel.get('user-email'));
-
-      return request.del(baseUrl + email + '/' + id, err => {
-        if (err) {
-          return next(err);
-        }
-        return super.saveValues(req, res, next);
-      });
+    const email = encodeEmail(req.sessionModel.get('user-email'));
+    try {
+      const model = new Model();
+      const params = {
+        url: `${baseUrl}${email}/${id}`,
+        method: 'DELETE'
+      };
+      await model._request(params);
+    } catch (err) {
+      logger.error(`Error deleting data: ${err.message}`);
+      return next(new Error(err.body || ''));
```

Comment:
Hi @Rhodine-orleans-lindsay the try catch block looks good, just a small suggestion, maybe in line 36, if the err.body is undefined, maybe we can add some error message, like "error occurred while deleting the form", instead of an empty string.

Reply:
Thanks Adesh, I have added an error message now.